base/applications/network/tracert/tracert.cpp

```diff
@@ -153,23 +158,85 @@ Usage()
     OutputText(IDS_USAGE);
 }
 
-static ULONG
+static
+bool IsStringNumericW(
+    _In_z_ LPCWSTR String) 
+{
+    WORD charType = 0;
+
+    // check input parameters
+    if (String == NULL)
+    {
+        SetLastError(ERROR_INVALID_PARAMETER);
+        return false;
+    }
+
+    // process String
+    while (*String) 
+    {
+        // gather information about the character we are on
+        if (GetStringTypeW(CT_CTYPE1, String, 1, &charType)) 
+        {
+            // check if the character is classified as a digit
+            // if it is not a digit set the last error
+            // and return
+            if ((charType & C1_DIGIT) == false) 
+            {
+                SetLastError(ERROR_INVALID_DATA);
+                return false;
+            }
+        }
+        else 
+        {
+            // Error retrieving character type
+            SetLastError(ERROR_INVALID_DATA);
+            return false;
+        }
+        String++;
+    }
+
+    // String appears to only contain digits
+    SetLastError(ERROR_SUCCESS);
+    return true;
+}
+
+static bool
 GetULONG(
```

Comment:
Looking again at the code, I wonder why this GetULONG function has to be that much complicated.
Just call wcstoul as you did + the check you do after it should be more than sufficient.